go/vt/vtgate/executor.go

```diff
@@ -1607,15 +1607,14 @@ func (e *Executor) handlePrepare(ctx context.Context, safeSession *econtext.Safe
 
 	qr, err := plan.Instructions.GetFields(ctx, vcursor, bindVars)
 	logStats.ExecuteTime = time.Since(execStart)
-	var errCount uint64
 	if err != nil {
 		logStats.Error = err
-		errCount = 1 // nolint
+		plan.AddStats(1, time.Since(logStats.StartTime), logStats.ShardQueries, 0, 0, 1)
```

Comment:
Other parts of VTGate are using this same approach for incrementing the err count of plans:
1. `0` rows affected
2. `0` rows returned
3. `1` err count